pkg/helm/informers.go

```diff
@@ -0,0 +1,178 @@
+	renderedInformers := []appstatetypes.StatusInformerString{}
+
+	// render status informers
+	for _, informer := range kotsKinds.KotsApplication.Spec.StatusInformers {
+		renderedInformer, err := builder.String(informer)
+		if err != nil {
+			logger.Errorf("failed to render status informer: %v\n", err)
+			continue
+		}
+		if renderedInformer == "" {
+			continue
+		}
+		renderedInformers = append(renderedInformers, appstatetypes.StatusInformerString(renderedInformer))
+	}
+
+	if identitydeploy.IsEnabled(kotsKinds.Identity, kotsKinds.IdentityConfig) {
+		renderedInformers = append(renderedInformers, appstatetypes.StatusInformerString(fmt.Sprintf("deployment/%s", identitytypes.DeploymentName(a.Slug))))
+	}
+
+	if len(renderedInformers) > 0 {
+		applyAppInformers(a.ID, sequence, renderedInformers, namespace)
+	} else {
+		// no informers, set state to ready
+		defaultReadyState := appstatetypes.ResourceStates{
+			{
+				Kind:      "EMPTY",
+				Name:      "EMPTY",
+				Namespace: "EMPTY",
+				State:     appstatetypes.StateReady,
+			},
+		}
+
+		release := GetHelmApp(a.ID)
+		appStatus := appstatetypes.AppStatus{
+			AppID:          a.ID,
+			ResourceStates: defaultReadyState,
+			Sequence:       release.GetCurrentSequence(),
+			UpdatedAt:      time.Now(),
+		}
+		release.Status = appStatus
+		release.Status.State = appstatetypes.GetState(defaultReadyState)
+		AddHelmApp(release.Release.Name, release)
+	}
+
+	return nil
+}
+
+func applyAppInformers(appID string, sequence int64, informerStrings []appstatetypes.StatusInformerString, namespace string) {
+	logger.Infof("received an inform event for appID (%s) sequence (%d) with informerStrings (%+v)\n", appID, sequence, informerStrings)
+
+	var informers []appstatetypes.StatusInformer
+	for _, str := range informerStrings {
+		informer, err := str.Parse()
+		if err != nil {
+			logger.Infof("failed to parse informer %s: %s", str, err.Error())
+			continue // don't stop
+		}
+		informers = append(informers, informer)
+	}
+	if len(informers) > 0 {
+		if mon, ok := monitorMap[namespace]; ok {
+			mon.Apply(appID, sequence, informers)
+		}
+	}
+}
+
+func runAppStateMonitor(monitor *appstate.Monitor) error {
+	m := map[string]func(f func()){}
+	hash := map[string]uint64{}
+	var mtx sync.Mutex
+
+	for appStatus := range monitor.AppStatusChan() {
+		throttled, ok := m[appStatus.AppID]
+		if !ok {
+			throttled = util.NewThrottle(time.Second)
+			m[appStatus.AppID] = throttled
+		}
+		throttled(func() {
+			mtx.Lock()
+			lastHash := hash[appStatus.AppID]
+			nextHash, _ := hashstructure.Hash(appStatus, nil)
+			hash[appStatus.AppID] = nextHash
+			mtx.Unlock()
+			if lastHash != nextHash {
+				b, _ := json.Marshal(appStatus)
+				logger.Infof("Updating app status %s", b)
+			}
+
+			app := GetHelmApp(appStatus.AppID)
+			app.Status = appStatus
+			app.Status.State = appstatetypes.GetState(appStatus.ResourceStates)
+			AddHelmApp(app.Release.Name, app)
```

Comment:
Let's not call this here.  Modifying the app is sufficient because it's a pointer.